frontend/src/logic/schedule/data-row.logic.ts

```diff
@@ -59,4 +63,37 @@ export class DataRow {
   public findValues(...args: string[]): string[] {
     return args.map((arg) => this.findValue(arg));
   }
+  //#endregion
+
+  //# row check region
+  private checkShiftRowPattern(): boolean {
+    const containsNotEmptyKey = this.data[0] != null && this.data[0] != "";
+
+    let shiftCodesList = Object.keys(ShiftCode)
+        .map(k => ShiftCode[k as any])
+    let containsAnyShiftCode = false;
+    shiftCodesList.forEach(shift => {
+      if(this.data.includes(shift)){
+        containsAnyShiftCode = true;
+      }
+    })
+
+    const dataLen = this.data.length
+    const hoursCellsNumber = 3
```

Comment:
Nie jestem pewien co do tego czy ma duży sens wprowadzać taki constraint. Bo niewiadomo, czy zawsze tam będzie dokładnie taka ilość komórek z godzinami. Moim zdaniem, całkiem wystarczy sprawdzać tylko na to czy jest w wierszu jest chociażby jedna zmiana

Reply:
Dorzucę TODO, żeby to zwalidować jak pojawią się nowe harmonogramy

Reply:
Dobra